src/Asset/ExecutionEngine/AutomationAction/Messenger/Handler/CsvCreationHandler.php

```diff
@@ -59,87 +52,47 @@ public function __construct(
     public function __invoke(CsvCreationMessage $message): void
     {
         $jobRun = $this->getJobRun($message);
-
-        $validatedParameters = $this->validateJobParameters(
-            $message,
-            $jobRun,
-            $this->userResolver
+        $settings = $this->extractConfigFieldFromJobStepConfig($message, Csv::JOB_STEP_CONFIG_SETTINGS->value);
+        $configuration = $this->gridService->getConfigurationFromArray(
+            $this->extractConfigFieldFromJobStepConfig($message, Csv::JOB_STEP_CONFIG_CONFIGURATION->value)
         );
-
-        if ($validatedParameters instanceof AbortActionData) {
-            $this->abort($validatedParameters);
-        }
-
-        $context = $jobRun->getContext();
-
-        if (!array_key_exists(ZipServiceInterface::ASSETS_INDEX, $context)) {
-            $this->abort(
-                $this->getAbortData(
-                    Config::NO_ASSETS_FOUND_FOR_JOB_RUN->value,
-                    [
-                        'jobRunId' => $jobRun->getId(),
-                    ]
-                )
-            );
-        }
-
-        $jobAsset = $validatedParameters->getSubject();
-
-        if (!in_array($jobAsset->getId(), $context[ZipServiceInterface::ASSETS_INDEX], true)) {
+        if (!isset($jobRun->getContext()[Csv::ASSET_EXPORT_DATA->value])) {
             $this->abort($this->getAbortData(
-                Config::ELEMENT_PERMISSION_MISSING_MESSAGE->value,
-                [
-                    'userId' => $jobRun->getOwnerId(),
-                    'permission' => ElementPermissions::VIEW_PERMISSION,
-                    'type' => ucfirst($jobAsset->getType()),
-                    'id' => $jobAsset->getId(),
-                ],
+                Config::CSV_CREATION_FAILED_MESSAGE->value,
+                ['message' => 'Asset export data not found in job run context']
             ));
         }
-
-        $settings = $this->extractConfigFieldFromJobStepConfig($message, 'settings');
-        $configuration = $this->gridService->getConfigurationFromArray(
-            $this->extractConfigFieldFromJobStepConfig($message, 'configuration')
-        );
-
-        $csv = $this->csvService->getCsvFile($jobRun->getId(), $configuration, $settings);
-
-        if (!$csv) {
+        $assetData = $jobRun->getContext()[Csv::ASSET_EXPORT_DATA->value];
+
+        try {
+            $this->csvService->createCsvFile(
+                $jobRun->getId(),
+                $settings[Csv::SETTINGS_DELIMITER->value] ?? ',',
```

Comment:
Should we somewhere configure a default delimiter?

Reply:
Not sure, this should come from the grid config afaik ... what would you suggest?

Reply:
It's about if there is nothing from the config. So the ?? ',' path. 
If we should move the ',' somewhere in a config.